### PR TITLE
tests: consolidate proxy tests

### PR DESCRIPTION
**Pull request by lucasmoura**
## Why is this needed?
We are better consolidating our proxy tests by removing whole scenarios that only test setting the proxy by writing directly in the user-config.json file while also removing some redundant scenarios on the proxy tests

## Test Steps
Verify that the modified proxy tests are still working and that we are not loosing any coverage regarding the tests I removed

---

- [ ] *(un)check this to re-run the checklist action*

### CODE REVIEW

**Review by dheyay — APPROVED** [review]
:+1: :+1:

**Review by renanrodrigo — APPROVED** [review]

File: features/proxy_config.feature [hunk 1]
```diff
@@ -146,6 +146,34 @@ Feature: Proxy configuration
       Acquire::http::Proxy::esm.ubuntu.com \".*\";
       Acquire::https::Proxy::esm.ubuntu.com \".*\";
       """
+    When I run `pro config unset ua_apt_http_proxy` with sudo
+    And I run `pro config unset ua_apt_https_proxy` with sudo
+    And I run `pro refresh config` with sudo
```
Comment (orndorffgrant) [inline features/proxy_config.feature:151]: I don't think we need this line
```suggestion
```

**Review by orndorffgrant — APPROVED** [review]
Two tiny nits so approving

File: features/proxy_config.feature [hunk 2]
```diff
@@ -146,6 +146,34 @@ Feature: Proxy configuration
...
+      }
+      """
+    And I run `pro refresh config` with sudo
+    Then I will see the following on stdout:
+      """
+      Setting snap proxy
+      Successfully processed your pro configuration.
+      """
+    When I run `truncate -s 0 /var/log/squid/access.log` `with sudo` on the `proxy` machine
+    # We need this for the route command
```
[the remark this answers is not in the document]
Comment (lucasmoura) [reply features/proxy_config.feature:169]: indeed

**Comment by lucasmoura** [comment]
@orndorffgrant updated

**Pull request merged**